test/OpenBCISample-test.js

```diff
@@ -30,76 +30,79 @@ var sampleBuf = samplePacket();
 
 
 describe('openBCISample',function() {
-    describe('#convertPacketToSample', function() {
-        it('should have the correct start byte', function() {
-            var sample = openBCISample.convertPacketToSample(sampleBuf);
-            assert.equal(k.OBCIByteStart,sample.startByte);
-        });
-        it('should have the correct stop byte', function() {
-            var sample = openBCISample.convertPacketToSample(sampleBuf);
-            assert.equal(k.OBCIByteStop,sample.stopByte);
+    var channelScaleFactor = 4.5 / 24 / (Math.pow(2,23) - 1);
+    describe('#parseRawPacket', function() {
+        it('should fulfill promise', function() {
+            return openBCISample.parseRawPacket(sampleBuf).should.be.fulfilled;
         });
         it('should have the correct sample number', function() {
-            var sample = openBCISample.convertPacketToSample(sampleBuf);
-            assert.equal(0x45,sample.sampleNumber);
-        });
-        it('all the channels should have the same number value as their (index + 1) * scaleFactor', function() {
-            var sample = openBCISample.convertPacketToSample(sampleBuf);
-            for(var i = 0;i < k.OBCINumberOfChannelsDefault;i++) {
-                //console.log(openBCISample.scaleFactorChannel * i);
-                assert.equal(sample.channelData[i],openBCISample.scaleFactorChannel * (i+1));
-            }
-        });
-        it('all the auxs should have the same number value as their index * scaleFactor', function() {
-            var sample = openBCISample.convertPacketToSample(sampleBuf);
-            for(var i = 0;i < 3;i++) {
-                assert.equal(sample.auxData[i],openBCISample.scaleFactorAux * i);
-            }
-        });
-        it('check to see if negative numbers work on channel data',function() {
+            return openBCISample.parseRawPacket(sampleBuf).should.eventually.have.property('sampleNumber').equal(0x45);
+        });
+        it('all the channels should have the same number value as their (index + 1) * scaleFactor', function(done) {
```

Comment:
Check this:
```js
var samplePacket = function () {
    var byteSample = 0x45;
    var buffy = new Buffer([0xA0,byteSample,0,0,1,0,0,2,0,0,3,0,0,4,0,0,5,0,0,6,0,0,7,0,0,8,0,0,0,1,0,2, 0xC0]);
    return buffy;
};
```
So look an see how 5 bytes in we have a `1` then three more we have a `2` all the way to `8`